libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PagerAdapterObserver.kt

```diff
@@ -0,0 +1,63 @@
+package org.wordpress.android.imageeditor.preview
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy
+
+/**
```

Comment:
> I'm new to the ViewPager2. Moreover, I haven't used any ViewPager with a Tablayout for a long time. I'm sorry if the questions don't make sense.

I'm new to `ViewPager2` as well :D. Please don't feel sorry, all thoughts are welcome that can improve the implementation 🙌. 

I think flickering was because of all tabs getting removed and then re-added in the default adapter observer whenever adapter refreshed. I tried to keep the tabs and just update the url to bring a little improvement. It can be further improved, but I'm not sure if we should do it now or later after we've finished higher priority tasks. Utilising `onItemRangeChanged` and `DiffUtil` seems like a promising solution. Thanks for sharing your thoughts 🙏.

Reply:
Thanks for the answer! ;) I agree that we can get back to optimizations later.